massa-pool/src/tests/scenario.rs

```diff
@@ -183,6 +184,172 @@ async fn test_pool() {
+                let (op, thread) = create_executesc(expire_period, fee, 100, 1); // Only the fee determines the rentability
+                let id = op.verify_integrity().unwrap();
+
+                let mut ops = OperationHashMap::default();
+                ops.insert(id, op.clone());
+
+                pool_command_sender
+                    .add_operations(ops.clone())
+                    .await
+                    .unwrap();
+
+                let newly_added = match protocol_controller
+                    .wait_command(250.into(), op_filter)
+                    .await
+                {
+                    Some(ProtocolCommand::PropagateOperations(ops)) => ops,
+                    Some(_) => panic!("unexpected protocol command"),
+                    None => panic!("unexpected timeout reached"),
+                };
+                assert_eq!(
+                    newly_added.keys().copied().collect::<Vec<_>>(),
+                    ops.keys().copied().collect::<Vec<_>>()
+                );
+
+                // duplicate
+                pool_command_sender
+                    .add_operations(ops.clone())
+                    .await
+                    .unwrap();
+
+                if let Some(cmd) = protocol_controller
+                    .wait_command(250.into(), op_filter)
+                    .await
+                {
+                    panic!("unexpected protocol command {:?}", cmd)
+                };
+
+                thread_tx_lists[thread as usize].push((id, op, start_period..=expire_period));
+            }
+            // sort from bigger fee to smaller and truncate
+            for lst in thread_tx_lists.iter_mut() {
+                lst.reverse();
+                lst.truncate(*max_pool_size_per_thread as usize);
+            }
+
+            // checks ops for thread 0 and 1 and various periods
+            for thread in 0u8..=1 {
+                for period in 0u64..70 {
+                    let target_slot = Slot::new(period, thread);
+                    let max_count = 3;
+                    let res = pool_command_sender
+                        .get_operation_batch(
+                            target_slot,
+                            OperationHashSet::default(),
+                            max_count,
+                            10000,
+                        )
+                        .await
+                        .unwrap();
+                    assert!(res
+                        .iter()
+                        .map(|(id, op, _)| (id, op.to_bytes_compact().unwrap()))
+                        .eq(thread_tx_lists[target_slot.thread as usize]
+                            .iter()
+                            .filter(|(_, _, r)| r.contains(&target_slot.period))
+                            .take(max_count)
+                            .map(|(id, op, _)| (id, op.to_bytes_compact().unwrap()))));
+                }
+            }
+            // op ending before or at period 45 should be discarded
```

Comment:
Here it would be clearer to describe what you are doing(discarding ops?), as opposed to using "should", because that makes it read like you are about to make assertions and instead you are manipulating the test data.